Assets/Scripts/EnemyController.cs

```diff
@@ -150,27 +164,65 @@ private void MoveToNextPoint()
          * the NavMeshAgent behaves in a physical manner (inertia) */
         if (!_wanderAI.IsInArea(wanderArea, transform.position))
         {
-            _agent.SetDestination(wanderArea.GetCenterPosition());
+            Move(wanderArea.GetCenterPosition());
         }
         else
         {
+            // don't use SetDestination but set calculated path
             NavMeshPath wanderPath = _wanderAI.GetNextWanderPath(wanderArea, _agent, moveLayer.value);
             _agent.SetPath(wanderPath);
+            _agent.isStopped = false;
         }
     }
+    
+    /// <summary>
+    /// Faces the target. Returns true if facing the target.
+    /// </summary>
+    /// <param name="target">The target to face</param>
+    /// <param name="shouldTurn">Whether the object should turn to the target or not</param>
+    /// <returns>Whether the object is facing the target</returns>
+    public bool FaceTarget(GameObject target, bool shouldTurn)
+    {
+        float f;
+        return FaceTarget(target, shouldTurn, out f);
```

Comment:
If you don't wanna use an out variable, you can simply use `out _` to discard the result

Reply:
forgot that here as well